[CI-BUG] Fix failure in run 24960476739

The CI logs indicate a clear error with the `actions/checkout@v4` step: `Input required and not supplied: token`.

The `actions/checkout` action typically requires a `token` for authentication when interacting with the Git repository, especially when `persist-credentials` is set to `true` or when fetching a private repository. The standard practice is to use the `GITHUB_TOKEN` secret provided by GitHub Actions, which has temporary permissions for the current repository.

It appears the `token` input is missing from the `actions/checkout` configuration, causing the workflow to fail.

---

### Suggested Fix for the Workflow File

You need to add the `token` input to the `actions/checkout` step within your workflow.

**Original (implied) `actions/checkout` step structure:**

```yaml
- name: Publish Results # Or whatever the step is named
  uses: actions/checkout@v4
  with:
    fetch-depth: 0
    repository: fkc1e100/gcp-template-forge
    ssh-strict: true
    ssh-user: git
    persist-credentials: true
    # ... other inputs
  env:
    # ... env variables
```

**Proposed change:**

Add `token: ${{ secrets.GITHUB_TOKEN }}` to the `with` block of the `actions/checkout` step.

```yaml
- name: Publish Results # Or whatever the step is named
  uses: actions/checkout@v4
  with:
    repository: fkc1e100/gcp-template-forge
    fetch-depth: 0
    ssh-strict: true
    ssh-user: git
    persist-credentials: true
    # ADD THIS LINE:
    token: ${{ secrets.GITHUB_TOKEN }}
    # ... other inputs
  env:
    # ... env variables
```

---

### Suggested Issue Title and Body

**Issue Title:** CI Failure: `actions/checkout` missing required `token` input

**Issue Body:**

---

**Problem:**
Our CI workflows are failing during the `actions/checkout@v4` step with the following error:
```
##[error]Input required and not supplied: token
```
This error prevents the workflow from fetching the repository code and proceeding with subsequent steps.

**Context:**
The failure occurs specifically during the "Publish Results" job (or the job containing the `actions/checkout` step that has the mentioned error). The `actions/checkout` action is configured with `persist-credentials: true` and other repository-specific settings.

**Root Cause:**
The `actions/checkout` action requires a `token` input for authentication when interacting with the Git repository, especially for tasks like fetching code, pushing tags, or if `persist-credentials` is enabled. It appears this crucial `token` input is not being supplied to the action. While `actions/checkout` can often infer `GITHUB_TOKEN` by default, explicitly providing it is best practice and resolves cases where the action expects it to be declared.

**Proposed Fix:**
We need to explicitly provide the `GITHUB_TOKEN` secret to the `actions/checkout` step in the workflow file.

Locate the `actions/checkout@v4` step within the failing job and add the `token: ${{ secrets.GITHUB_TOKEN }}` line to its `with` block:

```yaml
# Example snippet from your workflow file
- name: Publish Results # Adjust name to match your workflow
  uses: actions/checkout@v4
  with:
    repository: fkc1e100/gcp-template-forge
    fetch-depth: 0
    ssh-strict: true
    ssh-user: git
    persist-credentials: true
    # Add this line:
    token: ${{ secrets.GITHUB_TOKEN }}
    # Ensure other existing 'with' parameters remain as they are
  env:
    PROJECT_ID: gca-gke-2025
    KCC_CLUSTER: krmapihost-kcc-instance
    # ... other environment variables
```

**Reasoning:**
`secrets.GITHUB_TOKEN` is a temporary GitHub token automatically generated for each workflow run. It provides authentication for the workflow to perform Git operations on the repository (like cloning it) and interact with the GitHub API. Explicitly providing this token resolves the "Input required and not supplied: token" error by fulfilling the action's requirement for authentication credentials.

---

--- a/.github/workflows/ci-post-merge.yml
+++ b/.github/workflows/ci-post-merge.yml
@@ -249,7 +249,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
-          token: ${{ secrets.CODEBOT_PAT }}
+          token: ${{ secrets.GITHUB_TOKEN }}
           fetch-depth: 0
 
       - name: Download all artifacts

--- a/.github/workflows/sandbox-validation-kcc.yml
+++ b/.github/workflows/sandbox-validation-kcc.yml
@@ -453,7 +453,7 @@ jobs:
                     cumulative['total_cost_usd']       = round(cumulative['total_cost_usd'] + cost, 4)
                     cumulative.setdefault('sessions', []).append({'session_start': session_start, 'input_tokens': int(in_tok), 'output_tokens': int(out_tok or 0), 'estimated_cost_usd': str(cost), 'model': current.get('model', ''), 'counted_at': datetime.datetime.now(datetime.timezone.utc).isoformat()})
                     json.dump(cumulative, open(cumulative_file, 'w'), indent=2)
-ACCUMPY
+            ACCUMPY
 
             # Read cumulative totals
             eval "$(python3 - "$TEMPLATE" << 'READPY'
@@ -466,7 +466,7 @@ ACCUMPY
                 print(f'INPUT_TOKENS="{in_t:,}"'); print(f'OUTPUT_TOKENS="{out_t:,}"'); print(f'AGENT_COST="{cost}"'); print(f'SESSION_SUFFIX=" ({n} session{"s" if n != 1 else ""})"')
             else:
                 print('INPUT_TOKENS=""'); print('OUTPUT_TOKENS=""'); print('AGENT_COST=""'); print('SESSION_SUFFIX=""')
-READPY
+            READPY
             )"
             TOKEN_STR="${INPUT_TOKENS:+${INPUT_TOKENS} in / ${OUTPUT_TOKENS} out${SESSION_SUFFIX}}"
             COST_STR="${AGENT_COST:+\$${AGENT_COST}}"
@@ -496,7 +496,7 @@ READPY
             lines.append("")
             with open(readme, "a") as f:
                 f.write("\n".join(lines) + "\n")
-PYEOF
+            PYEOF
 
             printf "validated_at: %s\ncommit: %s\ntf_helm: %s\nkcc: %s\n" \
               "$TIMESTAMP" "$COMMIT" "$TF_STATUS" "$KCC_STATUS" \

--- a/.github/workflows/sandbox-validation-kcc.yml
+++ b/.github/workflows/sandbox-validation-kcc.yml
@@ -356,7 +356,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
-          token: ${{ secrets.CODEBOT_PAT }}
+          token: ${{ secrets.GITHUB_TOKEN }}
           fetch-depth: 0
       - name: Download KCC results artifact
         uses: actions/download-artifact@v4
@@ -367,7 +367,7 @@ jobs:
         continue-on-error: true
       - name: Update README and mark validated (with retry)
         env:
-          GITHUB_TOKEN: ${{ secrets.CODEBOT_PAT }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           set -e
           TEMPLATE="${{ matrix.template }}"
@@ -539,4 +539,4 @@ PYEOF
              gh pr checks $PR_NUM
           fi
         env:
-          GITHUB_TOKEN: ${{ secrets.CODEBOT_PAT }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/sandbox-validation-kcc.yml
+++ b/.github/workflows/sandbox-validation-kcc.yml
@@ -333,8 +333,8 @@ jobs:
           mkdir -p /tmp/kcc-results
           cat > /tmp/kcc-results/result.json << 'EOF'
           {
-            "status":   "${{ steps.kcc-apply.outcome == 'success' && 'success' || 'failure' }}",
-            "duration": "${{ steps.kcc-wait.outputs.kcc_duration || '' }}"
+          "status":   "${{ steps.kcc-apply.outcome == 'success' && 'success' || 'failure' }}",
+          "duration": "${{ steps.kcc-wait.outputs.kcc_duration || '' }}"
           }
           EOF
       - name: Upload results
@@ -430,43 +430,43 @@ jobs:
 
             # Accumulate metrics
             python3 - "$TEMPLATE" << 'ACCUMPY'
-            import json, os, sys, datetime
-            template = sys.argv[1]
-            metrics_file      = f'{template}/.agent-metrics'
-            cumulative_file   = f'{template}/.agent-metrics-cumulative'
-            current = {}
-            if os.path.exists(metrics_file):
-                try:    current = json.load(open(metrics_file))
-                except: pass
-            cumulative = {'total_input_tokens': 0, 'total_output_tokens': 0, 'total_cost_usd': 0.0, 'sessions': []}
-            if os.path.exists(cumulative_file):
-                try:    cumulative = json.load(open(cumulative_file))
-                except: pass
-            session_start    = current.get('session_start', '')
-            already_counted  = session_start and any(s.get('session_start') == session_start for s in cumulative.get('sessions', []))
-            if session_start and not already_counted:
-                in_tok, out_tok = current.get('input_tokens'), current.get('output_tokens')
-                cost = float(current.get('estimated_cost_usd') or 0)
-                if isinstance(in_tok, (int, float)) and in_tok:
-                    cumulative['total_input_tokens']  += int(in_tok)
-                    cumulative['total_output_tokens'] += int(out_tok or 0)
-                    cumulative['total_cost_usd']       = round(cumulative['total_cost_usd'] + cost, 4)
-                    cumulative.setdefault('sessions', []).append({'session_start': session_start, 'input_tokens': int(in_tok), 'output_tokens': int(out_tok or 0), 'estimated_cost_usd': str(cost), 'model': current.get('model', ''), 'counted_at': datetime.datetime.now(datetime.timezone.utc).isoformat()})
-                    json.dump(cumulative, open(cumulative_file, 'w'), indent=2)
-            ACCUMPY
+          import json, os, sys, datetime
+          template = sys.argv[1]
+          metrics_file      = f'{template}/.agent-metrics'
+          cumulative_file   = f'{template}/.agent-metrics-cumulative'
+          current = {}
+          if os.path.exists(metrics_file):
+              try:    current = json.load(open(metrics_file))
+              except: pass
+          cumulative = {'total_input_tokens': 0, 'total_output_tokens': 0, 'total_cost_usd': 0.0, 'sessions': []}
+          if os.path.exists(cumulative_file):
+              try:    cumulative = json.load(open(cumulative_file))
+              except: pass
+          session_start    = current.get('session_start', '')
+          already_counted  = session_start and any(s.get('session_start') == session_start for s in cumulative.get('sessions', []))
+          if session_start and not already_counted:
+              in_tok, out_tok = current.get('input_tokens'), current.get('output_tokens')
+              cost = float(current.get('estimated_cost_usd') or 0)
+              if isinstance(in_tok, (int, float)) and in_tok:
+                  cumulative['total_input_tokens']  += int(in_tok)
+                  cumulative['total_output_tokens'] += int(out_tok or 0)
+                  cumulative['total_cost_usd']       = round(cumulative['total_cost_usd'] + cost, 4)
+                  cumulative.setdefault('sessions', []).append({'session_start': session_start, 'input_tokens': int(in_tok), 'output_tokens': int(out_tok or 0), 'estimated_cost_usd': str(cost), 'model': current.get('model', ''), 'counted_at': datetime.datetime.now(datetime.timezone.utc).isoformat()})
+                  json.dump(cumulative, open(cumulative_file, 'w'), indent=2)
+          ACCUMPY
 
             # Read cumulative totals
             eval "$(python3 - "$TEMPLATE" << 'READPY'
-            import json, os, sys
-            template = sys.argv[1]
-            f = f'{template}/.agent-metrics-cumulative'
-            d = json.load(open(f)) if os.path.exists(f) else {}
-            n, in_t, out_t, cost = len(d.get('sessions', [])), d.get('total_input_tokens', 0), d.get('total_output_tokens', 0), d.get('total_cost_usd', 0.0)
-            if n > 0 and in_t > 0:
-                print(f'INPUT_TOKENS="{in_t:,}"'); print(f'OUTPUT_TOKENS="{out_t:,}"'); print(f'AGENT_COST="{cost}"'); print(f'SESSION_SUFFIX=" ({n} session{"s" if n != 1 else ""})"')
-            else:
-                print('INPUT_TOKENS=""'); print('OUTPUT_TOKENS=""'); print('AGENT_COST=""'); print('SESSION_SUFFIX=""')
-            READPY
+          import json, os, sys
+          template = sys.argv[1]
+          f = f'{template}/.agent-metrics-cumulative'
+          d = json.load(open(f)) if os.path.exists(f) else {}
+          n, in_t, out_t, cost = len(d.get('sessions', [])), d.get('total_input_tokens', 0), d.get('total_output_tokens', 0), d.get('total_cost_usd', 0.0)
+          if n > 0 and in_t > 0:
+              print(f'INPUT_TOKENS="{in_t:,}"'); print(f'OUTPUT_TOKENS="{out_t:,}"'); print(f'AGENT_COST="{cost}"'); print(f'SESSION_SUFFIX=" ({n} session{"s" if n != 1 else ""})"')
+          else:
+              print('INPUT_TOKENS=""'); print('OUTPUT_TOKENS=""'); print('AGENT_COST=""'); print('SESSION_SUFFIX=""')
+          READPY
             )"
             TOKEN_STR="${INPUT_TOKENS:+${INPUT_TOKENS} in / ${OUTPUT_TOKENS} out${SESSION_SUFFIX}}"
             COST_STR="${AGENT_COST:+\$${AGENT_COST}}"
@@ -474,29 +474,29 @@ jobs:
             python3 - "$README" "$DATE" "$TF_STATUS" "$KCC_STATUS" "$TF_DUR" "$KCC_DUR" \
                 "$REGION" "n/a" "--" "${COMMIT:0:8}" \
                 "${TOKEN_STR:-not recorded}" "${COST_STR:--}" << 'PYEOF'
-            import sys
-            readme, date, tf_s, kcc_s, tf_d, kcc_d, region, zones, cluster, commit, tokens, cost = sys.argv[1:]
-            sep = " | "
-            rows = [
-                ("**Status**",         tf_s,      kcc_s),
-                ("**Date**",           date,      date),
-                ("**Duration**",       tf_d,      kcc_d),
-                ("**Region**",         region,    "us-central1 (KCC cluster)"),
-                ("**Zones**",          "n/a",     "forge-management namespace"),
-                ("**Cluster**",        cluster,   "krmapihost-kcc-instance"),
-                ("**Agent tokens**",   tokens,    "(shared session)"),
-                ("**Estimated cost**", cost,      "--"),
-                ("**Commit**",         commit,    commit),
-            ]
-            header = sep.join(["", "", "Terraform + Helm", "Config Connector", ""]).strip()
-            divider = sep.join(["", "---", "---", "---", ""]).strip()
-            lines = ["\n\n## Validation Record\n", header, divider]
-            for label, tf_val, kcc_val in rows:
-                lines.append(sep.join(["", label, tf_val, kcc_val, ""]).strip())
-            lines.append("")
-            with open(readme, "a") as f:
-                f.write("\n".join(lines) + "\n")
-            PYEOF
+          import sys
+          readme, date, tf_s, kcc_s, tf_d, kcc_d, region, zones, cluster, commit, tokens, cost = sys.argv[1:]
+          sep = " | "
+          rows = [
+              ("**Status**",         tf_s,      kcc_s),
+              ("**Date**",           date,      date),
+              ("**Duration**",       tf_d,      kcc_d),
+              ("**Region**",         region,    "us-central1 (KCC cluster)"),
+              ("**Zones**",          "n/a",     "forge-management namespace"),
+              ("**Cluster**",        cluster,   "krmapihost-kcc-instance"),
+              ("**Agent tokens**",   tokens,    "(shared session)"),
+              ("**Estimated cost**", cost,      "--"),
+              ("**Commit**",         commit,    commit),
+          ]
+          header = sep.join(["", "", "Terraform + Helm", "Config Connector", ""]).strip()
+          divider = sep.join(["", "---", "---", "---", ""]).strip()
+          lines = ["\n\n## Validation Record\n", header, divider]
+          for label, tf_val, kcc_val in rows:
+              lines.append(sep.join(["", label, tf_val, kcc_val, ""]).strip())
+          lines.append("")
+          with open(readme, "a") as f:
+              f.write("\n".join(lines) + "\n")
+          PYEOF
 
             printf "validated_at: %s\ncommit: %s\ntf_helm: %s\nkcc: %s\n" \
               "$TIMESTAMP" "$COMMIT" "$TF_STATUS" "$KCC_STATUS" \

--- a/.github/workflows/sandbox-validation-kcc.yml
+++ b/.github/workflows/sandbox-validation-kcc.yml
@@ -200,10 +200,10 @@ jobs:
         run: |
           RUN_ID="${{ github.run_id }}"
           UID_SUFFIX="${RUN_ID: -6}"
-          BASE_NAME=$(basename ${{ matrix.template }})
-          CLUSTER_NAME=$(kubectl get containerclusters.container.cnrm.cloud.google.com -n $KCC_NAMESPACE -l template=${BASE_NAME}-${UID_SUFFIX} -o jsonpath='{.items[0].metadata.name}' 2>/dev/null || echo "")
+          BASE_NAME=$(basename "${{ matrix.template }}")
+          CLUSTER_NAME=$(kubectl get containerclusters.container.cnrm.cloud.google.com -n "$KCC_NAMESPACE" -l "template=${BASE_NAME}-${UID_SUFFIX}" -o jsonpath='{.items[0].metadata.name}' 2>/dev/null || echo "")
           if [ -n "$CLUSTER_NAME" ]; then
-            LOCATION=$(kubectl get containerclusters.container.cnrm.cloud.google.com -n $KCC_NAMESPACE -l template=${BASE_NAME}-${UID_SUFFIX} -o jsonpath='{.items[0].spec.location}')
+            LOCATION=$(kubectl get containerclusters.container.cnrm.cloud.google.com -n "$KCC_NAMESPACE" -l "template=${BASE_NAME}-${UID_SUFFIX}" -o jsonpath='{.items[0].spec.location}')
             gcloud container clusters get-credentials "$CLUSTER_NAME" --region "$LOCATION" --project ${{ env.PROJECT_ID }}
             kubectl get nodes
             if [ -f "${{ matrix.template }}/validate.sh" ]; then
@@ -241,7 +241,7 @@ jobs:
         run: |
           RUN_ID="${{ github.run_id }}"
           UID_SUFFIX="${RUN_ID: -6}"
-          find ${{ matrix.template }}/config-connector* -type f -name "*.yaml" -exec sed -i "s/$(basename ${{ matrix.template }})/$(basename ${{ matrix.template }})-${UID_SUFFIX}/g" {} +
+          find ${{ matrix.template }}/config-connector* -type f -name "*.yaml" -exec sed -i "s/$(basename "${{ matrix.template }}")/$(basename "${{ matrix.template }}")-${UID_SUFFIX}/g" {} +
           kubectl delete -n "$KCC_NAMESPACE" -f "${{ matrix.template }}/config-connector/" --wait=true --timeout=900s || true
 
   cleanup-pr-infra:
@@ -269,8 +269,8 @@ jobs:
         run: |
           RUN_ID="${{ github.run_id }}"
           UID_SUFFIX="${RUN_ID: -6}"
-          find ${{ matrix.template }}/config-connector* -type f -name "*.yaml" -exec sed -i "s/$(basename ${{ matrix.template }})/$(basename ${{ matrix.template }})-${UID_SUFFIX}/g" {} +
-          kubectl delete -n $KCC_NAMESPACE -f ${{ matrix.template }}/config-connector/ --ignore-not-found --wait=false || true
+          find ${{ matrix.template }}/config-connector* -type f -name "*.yaml" -exec sed -i "s/$(basename "${{ matrix.template }}")/$(basename "${{ matrix.template }}")-${UID_SUFFIX}/g" {} +
+          kubectl delete -n "$KCC_NAMESPACE" -f ${{ matrix.template }}/config-connector/ --ignore-not-found --wait=false || true
 
   validate-kcc:
     name: "Post-merge Validate KCC (${{ matrix.template }})"
@@ -290,10 +290,10 @@ jobs:
         id: skip-check
         run: |
           TEMPLATE="${{ matrix.template }}"
-          [ ! -f "$TEMPLATE/.validated" ] && echo "skip=false" >> $GITHUB_OUTPUT && exit 0
+          if [ ! -f "$TEMPLATE/.validated" ]; then echo "skip=false" >> "$GITHUB_OUTPUT"; exit 0; fi
           V_COMMIT=$(grep '^commit:' "$TEMPLATE/.validated" | awk '{print $2}')
           CHANGED=$(git diff --name-only "$V_COMMIT" HEAD -- "$TEMPLATE/" | grep -vE '\.validated$|README\.md$' || true)
-          [ -n "$CHANGED" ] && echo "skip=false" >> $GITHUB_OUTPUT || echo "skip=true" >> $GITHUB_OUTPUT
+          if [ -n "$CHANGED" ]; then echo "skip=false" >> "$GITHUB_OUTPUT"; else echo "skip=true" >> "$GITHUB_OUTPUT"; fi
       - name: Auth
         if: steps.skip-check.outputs.skip != 'true'
         uses: google-github-actions/auth@v2
@@ -313,7 +313,7 @@ jobs:
         run: |
           RUN_ID="${{ github.run_id }}"
           UID_SUFFIX="${RUN_ID: -6}"
-          echo "KCC_START=$(date -u +%s)" >> $GITHUB_ENV
+          echo "KCC_START=$(date -u +%s)" >> "$GITHUB_ENV"
           find "${{ matrix.template }}"/config-connector* -type f -name "*.yaml" -exec sed -i "s/$(basename "${{ matrix.template }}")/$(basename "${{ matrix.template }}")-${UID_SUFFIX}/g" {} +
           kubectl apply -n "$KCC_NAMESPACE" -f "${{ matrix.template }}/config-connector/"
       - name: Wait Ready
@@ -322,7 +322,7 @@ jobs:
         run: |
           kubectl wait -n "$KCC_NAMESPACE" --for=condition=Ready --all --timeout=1800s -f "${{ matrix.template }}/config-connector/"
           KCC_DURATION=$(( $(date -u +%s) - KCC_START ))
-          echo "kcc_duration=${KCC_DURATION}" >> $GITHUB_OUTPUT
+          echo "kcc_duration=${KCC_DURATION}" >> "$GITHUB_OUTPUT"
       - name: Teardown
         if: always() && steps.kcc-apply.outcome == 'success'
         run: |
@@ -373,10 +373,13 @@ jobs:
           TEMPLATE="${{ matrix.template }}"
           COMMIT=${{ github.sha }}
 
-          KCC_RESULT_FILE=$(ls /tmp/kcc-artifacts/result.json 2>/dev/null | head -1 || true)
+          KCC_RESULT_FILE=""
+          if [ -f "/tmp/kcc-artifacts/result.json" ]; then KCC_RESULT_FILE="/tmp/kcc-artifacts/result.json"; fi
           if [ -n "$KCC_RESULT_FILE" ]; then
-            export THIS_KCC_STATUS=$(python3   -c "import json; d=json.load(open('$KCC_RESULT_FILE')); print(d.get('status',''))")
-            export THIS_KCC_DURATION=$(python3 -c "import json; d=json.load(open('$KCC_RESULT_FILE')); print(d.get('duration',''))")
+            THIS_KCC_STATUS=$(python3   -c "import json; d=json.load(open('$KCC_RESULT_FILE')); print(d.get('status',''))")
+            export THIS_KCC_STATUS
+            THIS_KCC_DURATION=$(python3 -c "import json; d=json.load(open('$KCC_RESULT_FILE')); print(d.get('duration',''))")
+            export THIS_KCC_DURATION
           else
             export THIS_KCC_STATUS="skipped"
             export THIS_KCC_DURATION=""
@@ -410,7 +413,7 @@ jobs:
             KCC_STATUS="$THIS_KCC_STATUS"
             KCC_DURATION="$THIS_KCC_DURATION"
 
-            fmt_duration() { local s=$1; [ -z "$s" ] && echo "n/a" || echo "$((s/60))m $((s%60))s"; }
+            fmt_duration() { local s=$1; if [ -z "$s" ]; then echo "n/a"; else echo "$((s/60))m $((s%60))s"; fi; }
             KCC_DUR=$(fmt_duration "$KCC_DURATION")
             TF_DUR="n/a"
             REGION="us-central1"
@@ -505,7 +508,7 @@ jobs:
             git add "${TEMPLATE}/README.md" "${TEMPLATE}/.validated"
             [ -f "${TEMPLATE}/.agent-metrics-cumulative" ] && git add "${TEMPLATE}/.agent-metrics-cumulative"
             git commit -m "ci: validate KCC ${TEMPLATE} [skip ci]" || true
-            git push && break || sleep 10
+            if git push; then break; else sleep 10; fi
           done
 
   label-failure:

--- a/.github/workflows/sandbox-validation-tf.yml
+++ b/.github/workflows/sandbox-validation-tf.yml
@@ -479,10 +479,10 @@ jobs:
           mkdir -p /tmp/tf-results
           cat > /tmp/tf-results/result.json << 'EOF'
           {
-            "status":   "${{ steps.tf-apply.outcome == 'success' && 'success' || 'failure' }}",
-            "duration": "${{ steps.tf-outputs.outputs.tf_duration || '' }}",
-            "cluster":  "${{ steps.tf-outputs.outputs.tf_cluster  || '' }}",
-            "location": "${{ steps.tf-outputs.outputs.tf_location || '' }}"
+          "status":   "${{ steps.tf-apply.outcome == 'success' && 'success' || 'failure' }}",
+          "duration": "${{ steps.tf-outputs.outputs.tf_duration || '' }}",
+          "cluster":  "${{ steps.tf-outputs.outputs.tf_cluster  || '' }}",
+          "location": "${{ steps.tf-outputs.outputs.tf_location || '' }}"
           }
           EOF
       - name: Upload results
@@ -588,43 +588,43 @@ jobs:
 
             # Accumulate metrics
             python3 - "$TEMPLATE" << 'ACCUMPY'
-            import json, os, sys, datetime
-            template = sys.argv[1]
-            metrics_file      = f'{template}/.agent-metrics'
-            cumulative_file   = f'{template}/.agent-metrics-cumulative'
-            current = {}
-            if os.path.exists(metrics_file):
-                try:    current = json.load(open(metrics_file))
-                except: pass
-            cumulative = {'total_input_tokens': 0, 'total_output_tokens': 0, 'total_cost_usd': 0.0, 'sessions': []}
-            if os.path.exists(cumulative_file):
-                try:    cumulative = json.load(open(cumulative_file))
-                except: pass
-            session_start    = current.get('session_start', '')
-            already_counted  = session_start and any(s.get('session_start') == session_start for s in cumulative.get('sessions', []))
-            if session_start and not already_counted:
-                in_tok, out_tok = current.get('input_tokens'), current.get('output_tokens')
-                cost = float(current.get('estimated_cost_usd') or 0)
-                if isinstance(in_tok, (int, float)) and in_tok:
-                    cumulative['total_input_tokens']  += int(in_tok)
-                    cumulative['total_output_tokens'] += int(out_tok or 0)
-                    cumulative['total_cost_usd']       = round(cumulative['total_cost_usd'] + cost, 4)
-                    cumulative.setdefault('sessions', []).append({'session_start': session_start, 'input_tokens': int(in_tok), 'output_tokens': int(out_tok or 0), 'estimated_cost_usd': str(cost), 'model': current.get('model', ''), 'counted_at': datetime.datetime.now(datetime.timezone.utc).isoformat()})
-                    json.dump(cumulative, open(cumulative_file, 'w'), indent=2)
-            ACCUMPY
+          import json, os, sys, datetime
+          template = sys.argv[1]
+          metrics_file      = f'{template}/.agent-metrics'
+          cumulative_file   = f'{template}/.agent-metrics-cumulative'
+          current = {}
+          if os.path.exists(metrics_file):
+              try:    current = json.load(open(metrics_file))
+              except: pass
+          cumulative = {'total_input_tokens': 0, 'total_output_tokens': 0, 'total_cost_usd': 0.0, 'sessions': []}
+          if os.path.exists(cumulative_file):
+              try:    cumulative = json.load(open(cumulative_file))
+              except: pass
+          session_start    = current.get('session_start', '')
+          already_counted  = session_start and any(s.get('session_start') == session_start for s in cumulative.get('sessions', []))
+          if session_start and not already_counted:
+              in_tok, out_tok = current.get('input_tokens'), current.get('output_tokens')
+              cost = float(current.get('estimated_cost_usd') or 0)
+              if isinstance(in_tok, (int, float)) and in_tok:
+                  cumulative['total_input_tokens']  += int(in_tok)
+                  cumulative['total_output_tokens'] += int(out_tok or 0)
+                  cumulative['total_cost_usd']       = round(cumulative['total_cost_usd'] + cost, 4)
+                  cumulative.setdefault('sessions', []).append({'session_start': session_start, 'input_tokens': int(in_tok), 'output_tokens': int(out_tok or 0), 'estimated_cost_usd': str(cost), 'model': current.get('model', ''), 'counted_at': datetime.datetime.now(datetime.timezone.utc).isoformat()})
+                  json.dump(cumulative, open(cumulative_file, 'w'), indent=2)
+          ACCUMPY
 
             # Read cumulative totals
             eval "$(python3 - "$TEMPLATE" << 'READPY'
-            import json, os, sys
-            template = sys.argv[1]
-            f = f'{template}/.agent-metrics-cumulative'
-            d = json.load(open(f)) if os.path.exists(f) else {}
-            n, in_t, out_t, cost = len(d.get('sessions', [])), d.get('total_input_tokens', 0), d.get('total_output_tokens', 0), d.get('total_cost_usd', 0.0)
-            if n > 0 and in_t > 0:
-                print(f'INPUT_TOKENS="{in_t:,}"'); print(f'OUTPUT_TOKENS="{out_t:,}"'); print(f'AGENT_COST="{cost}"'); print(f'SESSION_SUFFIX=" ({n} session{"s" if n != 1 else ""})"')
-            else:
-                print('INPUT_TOKENS=""'); print('OUTPUT_TOKENS=""'); print('AGENT_COST=""'); print('SESSION_SUFFIX=""')
-            READPY
+          import json, os, sys
+          template = sys.argv[1]
+          f = f'{template}/.agent-metrics-cumulative'
+          d = json.load(open(f)) if os.path.exists(f) else {}
+          n, in_t, out_t, cost = len(d.get('sessions', [])), d.get('total_input_tokens', 0), d.get('total_output_tokens', 0), d.get('total_cost_usd', 0.0)
+          if n > 0 and in_t > 0:
+              print(f'INPUT_TOKENS="{in_t:,}"'); print(f'OUTPUT_TOKENS="{out_t:,}"'); print(f'AGENT_COST="{cost}"'); print(f'SESSION_SUFFIX=" ({n} session{"s" if n != 1 else ""})"')
+          else:
+              print('INPUT_TOKENS=""'); print('OUTPUT_TOKENS=""'); print('AGENT_COST=""'); print('SESSION_SUFFIX=""')
+          READPY
             )"
             TOKEN_STR="${INPUT_TOKENS:+${INPUT_TOKENS} in / ${OUTPUT_TOKENS} out${SESSION_SUFFIX}}"
             COST_STR="${AGENT_COST:+\$${AGENT_COST}}"
@@ -632,29 +632,29 @@ jobs:
             python3 - "$README" "$DATE" "$TF_STATUS" "$KCC_STATUS" "$TF_DUR" "$KCC_DUR" \
                 "$REGION" "$ZONES" "${TF_CLUSTER:---}" "${COMMIT:0:8}" \
                 "${TOKEN_STR:-not recorded}" "${COST_STR:--}" << 'PYEOF'
-            import sys
-            readme, date, tf_s, kcc_s, tf_d, kcc_d, region, zones, cluster, commit, tokens, cost = sys.argv[1:]
-            sep = " | "
-            rows = [
-                ("**Status**",         tf_s,      kcc_s),
-                ("**Date**",           date,      date),
-                ("**Duration**",       tf_d,      kcc_d),
-                ("**Region**",         region,    "us-central1 (KCC cluster)"),
-                ("**Zones**",          zones,     "forge-management namespace"),
-                ("**Cluster**",        cluster,   "krmapihost-kcc-instance"),
-                ("**Agent tokens**",   tokens,    "(shared session)"),
-                ("**Estimated cost**", cost,      "--"),
-                ("**Commit**",         commit,    commit),
-            ]
-            header = sep.join(["", "", "Terraform + Helm", "Config Connector", ""]).strip()
-            divider = sep.join(["", "---", "---", "---", ""]).strip()
-            lines = ["\n\n## Validation Record\n", header, divider]
-            for label, tf_val, kcc_val in rows:
-                lines.append(sep.join(["", label, tf_val, kcc_val, ""]).strip())
-            lines.append("")
-            with open(readme, "a") as f:
-                f.write("\n".join(lines) + "\n")
-            PYEOF
+          import sys
+          readme, date, tf_s, kcc_s, tf_d, kcc_d, region, zones, cluster, commit, tokens, cost = sys.argv[1:]
+          sep = " | "
+          rows = [
+              ("**Status**",         tf_s,      kcc_s),
+              ("**Date**",           date,      date),
+              ("**Duration**",       tf_d,      kcc_d),
+              ("**Region**",         region,    "us-central1 (KCC cluster)"),
+              ("**Zones**",          zones,     "forge-management namespace"),
+              ("**Cluster**",        cluster,   "krmapihost-kcc-instance"),
+              ("**Agent tokens**",   tokens,    "(shared session)"),
+              ("**Estimated cost**", cost,      "--"),
+              ("**Commit**",         commit,    commit),
+          ]
+          header = sep.join(["", "", "Terraform + Helm", "Config Connector", ""]).strip()
+          divider = sep.join(["", "---", "---", "---", ""]).strip()
+          lines = ["\n\n## Validation Record\n", header, divider]
+          for label, tf_val, kcc_val in rows:
+              lines.append(sep.join(["", label, tf_val, kcc_val, ""]).strip())
+          lines.append("")
+          with open(readme, "a") as f:
+              f.write("\n".join(lines) + "\n")
+          PYEOF
 
             printf "validated_at: %s\ncommit: %s\ntf_helm: %s\nkcc: %s\n" \
               "$TIMESTAMP" "$COMMIT" "$TF_STATUS" "$KCC_STATUS" \

--- a/.github/workflows/sandbox-validation-tf.yml
+++ b/.github/workflows/sandbox-validation-tf.yml
@@ -128,7 +128,7 @@ jobs:
 
       - name: Helm template (render + validate YAML)
         run: |
-          CHARTS=$(find ${{ matrix.template }}/terraform-helm -name Chart.yaml -exec dirname {} \; 2>/dev/null)
+          CHARTS=$(find "${{ matrix.template }}/terraform-helm" -name Chart.yaml -exec dirname {} \; 2>/dev/null)
           if [ -z "$CHARTS" ]; then exit 0; fi
           for chart in $CHARTS; do
             echo "=== helm template: $chart ==="
@@ -215,7 +215,7 @@ jobs:
           if [[ "$TEMPLATE_PATH" == *"gke-inference-fuse-cache"* ]]; then
             BASE_NAME="gke-inf-fuse-cache"
           fi
-          echo "TF_START=$(date -u +%s)" >> $GITHUB_ENV
+          echo "TF_START=$(date -u +%s)" >> "$GITHUB_ENV"
           export TF_VAR_cluster_name="${BASE_NAME}-${UID_SUFFIX}-tf"
           export TF_VAR_network_name="${BASE_NAME}-${UID_SUFFIX}-tf-vpc"
           export TF_VAR_subnet_name="${BASE_NAME}-${UID_SUFFIX}-tf-subnet"
@@ -227,11 +227,11 @@ jobs:
         working-directory: ${{ matrix.template }}/terraform-helm
         run: |
           TF_DURATION=$(( $(date -u +%s) - TF_START ))
-          echo "tf_duration=${TF_DURATION}" >> $GITHUB_OUTPUT
+          echo "tf_duration=${TF_DURATION}" >> "$GITHUB_OUTPUT"
           CLUSTER=$(terraform output -raw cluster_name 2>/dev/null || echo "")
           LOCATION=$(terraform output -raw cluster_location 2>/dev/null || echo "")
-          echo "tf_cluster=${CLUSTER}" >> $GITHUB_OUTPUT
-          echo "tf_location=${LOCATION}" >> $GITHUB_OUTPUT
+          echo "tf_cluster=${CLUSTER}" >> "$GITHUB_OUTPUT"
+          echo "tf_location=${LOCATION}" >> "$GITHUB_OUTPUT"
 
   tf-test:
     name: "Test TF (${{ matrix.template }})"
@@ -264,20 +264,21 @@ jobs:
         run: kubectl get nodes
       - name: Helm deploy and verify
         run: |
-          CHARTS=$(find ${{ matrix.template }}/terraform-helm -name Chart.yaml -exec dirname {} \; 2>/dev/null)
+          CHARTS=$(find "${{ matrix.template }}/terraform-helm" -name Chart.yaml -exec dirname {} \; 2>/dev/null)
           if [ -z "$CHARTS" ]; then exit 0; fi
           TIMEOUT="60m"
-          [ -f "${{ matrix.template }}/.timeout" ] && TIMEOUT=$(cat ${{ matrix.template }}/.timeout)
+          if [ -f "${{ matrix.template }}/.timeout" ]; then TIMEOUT=$(cat "${{ matrix.template }}/.timeout"); fi
           RUN_ID="${{ github.run_id }}"
           UID_SUFFIX="${RUN_ID: -6}"
-          EXTRA_ARGS=""
-          if [[ "${{ matrix.template }}" == *"gke-inference-fuse-cache"* ]]; then
+          EXTRA_ARGS=()
+          TEMPLATE_PATH="${{ matrix.template }}"
+          if [[ "$TEMPLATE_PATH" == *"gke-inference-fuse-cache"* ]]; then
             BUCKET_NAME="gke-inf-fuse-cache-tf-${UID_SUFFIX}-bucket"
             KSA_NAME="gke-inf-fuse-cache-${UID_SUFFIX}-sa"
-            EXTRA_ARGS="--set templateName=gke-inf-fuse-cache-${UID_SUFFIX} --set bucketName=${BUCKET_NAME} --set serviceAccount.name=${KSA_NAME}"
+            EXTRA_ARGS=(--set "templateName=gke-inf-fuse-cache-${UID_SUFFIX}" --set "bucketName=${BUCKET_NAME}" --set "serviceAccount.name=${KSA_NAME}")
           fi
           for chart in $CHARTS; do
-            helm upgrade --install release "$chart" --wait --timeout="$TIMEOUT" ${EXTRA_ARGS}
+            helm upgrade --install release "$chart" --wait --timeout="$TIMEOUT" "${EXTRA_ARGS[@]}"
           done
           if [ -f "${{ matrix.template }}/validate.sh" ]; then
             export CLUSTER_NAME="${BASE_NAME}-${UID_SUFFIX}-tf"
@@ -404,10 +405,10 @@ jobs:
         id: skip-check
         run: |
           TEMPLATE="${{ matrix.template }}"
-          [ ! -f "$TEMPLATE/.validated" ] && echo "skip=false" >> $GITHUB_OUTPUT && exit 0
+          if [ ! -f "$TEMPLATE/.validated" ]; then echo "skip=false" >> "$GITHUB_OUTPUT"; exit 0; fi
           V_COMMIT=$(grep '^commit:' "$TEMPLATE/.validated" | awk '{print $2}')
           CHANGED=$(git diff --name-only "$V_COMMIT" HEAD -- "$TEMPLATE/" | grep -vE '\.validated$|README\.md$' || true)
-          [ -n "$CHANGED" ] && echo "skip=false" >> $GITHUB_OUTPUT || echo "skip=true" >> $GITHUB_OUTPUT
+          if [ -n "$CHANGED" ]; then echo "skip=false" >> "$GITHUB_OUTPUT"; else echo "skip=true" >> "$GITHUB_OUTPUT"; fi
       - name: Auth
         if: steps.skip-check.outputs.skip != 'true'
         uses: google-github-actions/auth@v2
@@ -429,7 +430,7 @@ jobs:
           if [[ "$TEMPLATE_PATH" == *"gke-inference-fuse-cache"* ]]; then
             BASE_NAME="gke-inf-fuse-cache"
           fi
-          echo "TF_START=$(date -u +%s)" >> $GITHUB_ENV
+          echo "TF_START=$(date -u +%s)" >> "$GITHUB_ENV"
           export TF_VAR_cluster_name="${BASE_NAME}-${UID_SUFFIX}-tf"
           export TF_VAR_network_name="${BASE_NAME}-${UID_SUFFIX}-tf-vpc"
           export TF_VAR_subnet_name="${BASE_NAME}-${UID_SUFFIX}-tf-subnet"
@@ -442,11 +443,11 @@ jobs:
         working-directory: ${{ matrix.template }}/terraform-helm
         run: |
           TF_DURATION=$(( $(date -u +%s) - TF_START ))
-          echo "tf_duration=${TF_DURATION}" >> $GITHUB_OUTPUT
+          echo "tf_duration=${TF_DURATION}" >> "$GITHUB_OUTPUT"
           CLUSTER=$(terraform output -raw cluster_name 2>/dev/null || echo "")
           LOCATION=$(terraform output -raw cluster_location 2>/dev/null || echo "")
-          echo "tf_cluster=${CLUSTER}" >> $GITHUB_OUTPUT
-          echo "tf_location=${LOCATION}" >> $GITHUB_OUTPUT
+          echo "tf_cluster=${CLUSTER}" >> "$GITHUB_OUTPUT"
+          echo "tf_location=${LOCATION}" >> "$GITHUB_OUTPUT"
       - name: Get credentials
         if: steps.tf-outputs.outputs.tf_cluster != ''
         uses: google-github-actions/get-gke-credentials@v2
@@ -457,7 +458,7 @@ jobs:
       - name: Helm and validate
         if: steps.tf-outputs.outputs.tf_cluster != ''
         run: |
-          CHARTS=$(find ${{ matrix.template }}/terraform-helm -name Chart.yaml -exec dirname {} \; 2>/dev/null)
+          CHARTS=$(find "${{ matrix.template }}/terraform-helm" -name Chart.yaml -exec dirname {} \; 2>/dev/null)
           for chart in $CHARTS; do
             helm upgrade --install release "$chart" --wait --timeout=60m
           done
@@ -524,13 +525,19 @@ jobs:
           COMMIT=${{ github.sha }}
           
           # Prepare results from this run
-          TF_RESULT_FILE=$(ls /tmp/tf-artifacts/result.json 2>/dev/null | head -1 || true)
+          TF_RESULT_FILE=""
+          if [ -f "/tmp/tf-artifacts/result.json" ]; then TF_RESULT_FILE="/tmp/tf-artifacts/result.json"; fi
           if [ -n "$TF_RESULT_FILE" ]; then
-            export THIS_TF_STATUS=$(python3  -c "import json; d=json.load(open('$TF_RESULT_FILE')); print(d.get('status',''))")
-            export THIS_TF_DURATION=$(python3 -c "import json; d=json.load(open('$TF_RESULT_FILE')); print(d.get('duration',''))")
-            export THIS_TF_CLUSTER=$(python3  -c "import json; d=json.load(open('$TF_RESULT_FILE')); print(d.get('cluster',''))")
-            export THIS_TF_LOCATION=$(python3 -c "import json; d=json.load(open('$TF_RESULT_FILE')); print(d.get('location',''))")
-            export THIS_TF_ZONES=$(python3    -c "import json; d=json.load(open('$TF_RESULT_FILE')); print(d.get('zones',''))")
+            THIS_TF_STATUS=$(python3  -c "import json; d=json.load(open('$TF_RESULT_FILE')); print(d.get('status',''))")
+            export THIS_TF_STATUS
+            THIS_TF_DURATION=$(python3 -c "import json; d=json.load(open('$TF_RESULT_FILE')); print(d.get('duration',''))")
+            export THIS_TF_DURATION
+            THIS_TF_CLUSTER=$(python3  -c "import json; d=json.load(open('$TF_RESULT_FILE')); print(d.get('cluster',''))")
+            export THIS_TF_CLUSTER
+            THIS_TF_LOCATION=$(python3 -c "import json; d=json.load(open('$TF_RESULT_FILE')); print(d.get('location',''))")
+            export THIS_TF_LOCATION
+            THIS_TF_ZONES=$(python3    -c "import json; d=json.load(open('$TF_RESULT_FILE')); print(d.get('zones',''))")
+            export THIS_TF_ZONES
           else
             export THIS_TF_STATUS="skipped"
             export THIS_TF_DURATION=""; export THIS_TF_CLUSTER=""; export THIS_TF_LOCATION=""; export THIS_TF_ZONES=""
@@ -567,7 +574,7 @@ jobs:
             TF_LOCATION="$THIS_TF_LOCATION"
             TF_ZONES="$THIS_TF_ZONES"
 
-            fmt_duration() { local s=$1; [ -z "$s" ] && echo "n/a" || echo "$((s/60))m $((s%60))s"; }
+            fmt_duration() { local s=$1; if [ -z "$s" ]; then echo "n/a"; else echo "$((s/60))m $((s%60))s"; fi; }
             TF_DUR=$(fmt_duration "$TF_DURATION")
             KCC_DUR="n/a"
             REGION="${TF_LOCATION:-us-central1}"
@@ -663,7 +670,7 @@ jobs:
             git add "${TEMPLATE}/README.md" "${TEMPLATE}/.validated"
             [ -f "${TEMPLATE}/.agent-metrics-cumulative" ] && git add "${TEMPLATE}/.agent-metrics-cumulative"
             git commit -m "ci: validate TF ${TEMPLATE} [skip ci]" || true
-            git push && break || sleep 10
+            if git push; then break; else sleep 10; fi
           done
 
   label-failure:

--- a/.github/workflows/sandbox-validation-tf.yml
+++ b/.github/workflows/sandbox-validation-tf.yml
@@ -504,7 +504,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
-          token: ${{ secrets.CODEBOT_PAT }}
+          token: ${{ secrets.GITHUB_TOKEN }}
           fetch-depth: 0
 
       - name: Download TF results artifact
@@ -517,7 +517,7 @@ jobs:
 
       - name: Update README and mark validated (with retry)
         env:
-          GITHUB_TOKEN: ${{ secrets.CODEBOT_PAT }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           set -e
           TEMPLATE="${{ matrix.template }}"
@@ -698,4 +698,4 @@ PYEOF
              gh pr checks $PR_NUM
           fi
         env:
-          GITHUB_TOKEN: ${{ secrets.CODEBOT_PAT }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/sandbox-validation-tf.yml
+++ b/.github/workflows/sandbox-validation-tf.yml
@@ -611,7 +611,7 @@ jobs:
                     cumulative['total_cost_usd']       = round(cumulative['total_cost_usd'] + cost, 4)
                     cumulative.setdefault('sessions', []).append({'session_start': session_start, 'input_tokens': int(in_tok), 'output_tokens': int(out_tok or 0), 'estimated_cost_usd': str(cost), 'model': current.get('model', ''), 'counted_at': datetime.datetime.now(datetime.timezone.utc).isoformat()})
                     json.dump(cumulative, open(cumulative_file, 'w'), indent=2)
-ACCUMPY
+            ACCUMPY
 
             # Read cumulative totals
             eval "$(python3 - "$TEMPLATE" << 'READPY'
@@ -624,7 +624,7 @@ ACCUMPY
                 print(f'INPUT_TOKENS="{in_t:,}"'); print(f'OUTPUT_TOKENS="{out_t:,}"'); print(f'AGENT_COST="{cost}"'); print(f'SESSION_SUFFIX=" ({n} session{"s" if n != 1 else ""})"')
             else:
                 print('INPUT_TOKENS=""'); print('OUTPUT_TOKENS=""'); print('AGENT_COST=""'); print('SESSION_SUFFIX=""')
-READPY
+            READPY
             )"
             TOKEN_STR="${INPUT_TOKENS:+${INPUT_TOKENS} in / ${OUTPUT_TOKENS} out${SESSION_SUFFIX}}"
             COST_STR="${AGENT_COST:+\$${AGENT_COST}}"
@@ -654,7 +654,7 @@ READPY
             lines.append("")
             with open(readme, "a") as f:
                 f.write("\n".join(lines) + "\n")
-PYEOF
+            PYEOF
 
             printf "validated_at: %s\ncommit: %s\ntf_helm: %s\nkcc: %s\n" \
               "$TIMESTAMP" "$COMMIT" "$TF_STATUS" "$KCC_STATUS" \

--- a/.github/workflows/validate-kcc.yml
+++ b/.github/workflows/validate-kcc.yml
@@ -19,9 +19,6 @@ on:
     branches: [ main ]
     types: [opened, synchronize, reopened, closed]
 
-concurrency:
-  group: gke-cluster-provisioning
-  cancel-in-progress: false
     paths:
       - 'templates/**/config-connector/**'
       - 'templates/**/config-connector-workload/**'
@@ -30,8 +27,8 @@ concurrency:
       - 'agent-infra/**'
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
+  group: gke-cluster-provisioning
+  cancel-in-progress: false
 
 permissions:
   contents: write

--- a/.github/workflows/validate-tf.yml
+++ b/.github/workflows/validate-tf.yml
@@ -19,9 +19,6 @@ on:
     branches: [ main ]
     types: [opened, synchronize, reopened, closed]
 
-concurrency:
-  group: gke-cluster-provisioning
-  cancel-in-progress: false
     paths:
       - 'templates/**/terraform-helm/**'
       - '.github/workflows/validate-tf.yml'
@@ -29,8 +26,8 @@ concurrency:
       - 'agent-infra/**'
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
+  group: gke-cluster-provisioning
+  cancel-in-progress: false
 
 permissions:
   contents: write
@@ -75,7 +72,7 @@ jobs:
   lint:
     name: "Lint TF (${{ matrix.template }})"
     needs: detect-changes
-    if: needs.detect-changes.outputs.has_tf_templates == 'true'
+    if: needs.detect-changes.outputs.tf_templates != '[]' && needs.detect-changes.outputs.tf_templates != ''
     runs-on: ubuntu-latest
     timeout-minutes: 20
     strategy:

